src/test/java/io/confluent/examples/streams/IntegrationTestUtils.java

```diff
@@ -277,4 +279,87 @@ private static boolean continueConsuming(final int messagesConsumed, final int m
         30000,
         "Expected values not found in WindowStore"); }
 
+  static <K, V> Map<K, V> drainTableOutput(final String outputTopic,
+                                           final TopologyTestDriver topologyTestDriver,
+                                           final Deserializer<K> keyDeserializer,
+                                           final Deserializer<V> valueDeserializer) {
+
+    final Map<K, V> results = new LinkedHashMap<>();
+    while (true) {
+      final ProducerRecord<K, V> record = topologyTestDriver.readOutput(outputTopic, keyDeserializer, valueDeserializer);
+      if (record == null) {
+        break;
+      } else {
+        results.put(record.key(), record.value());
+      }
+    }
+    return results;
+  }
+
+  static <K, V> void produceKeyValuesSynchronously(final String inputTopic,
+                                                   final List<KeyValue<K, V>> entities,
+                                                   final TopologyTestDriver topologyTestDriver,
+                                                   final Serializer<K> keySerializer,
+                                                   final Serializer<V> valueSerializer) {
+    for (final KeyValue<K, V> entity : entities) {
+      topologyTestDriver.pipeInput(new ConsumerRecord<>(
+        inputTopic,
+        0,
+        0,
+        0,
+        TimestampType.CREATE_TIME,
+        -1,
+        -1,
+        -1,
+        keySerializer.serialize(inputTopic, entity.key),
+        valueSerializer.serialize(inputTopic, entity.value)
+      ));
```

Comment:
Use constants, like `NO_TIMESTAMP`, `NO_CHECKSUM` and similar. For others (with no available consts, add comment?) -- Also, there are multiple constructors -- can't we use one with less parameters?

Reply:
Roger using constants.

Sadly, there's no constructor that lets you set topic, partition, and timestamp except for the full constructor.